sky/resources.py

```diff
@@ -275,7 +275,35 @@ def _set_region_zone(self, region: Optional[str],
         self._region, self._zone = self._cloud.validate_region_zone(
             region, zone)
 
-    def _try_validate_instance_type(self):
+    def get_valid_region_zones(self) -> List[Tuple[clouds.Region, clouds.Zone]]:
+        """Returns a list of (region, zone) that can provision this Resources"""
+        assert self.is_launchable()
+
+        if isinstance(self._cloud, clouds.GCP):
```

Comment:
nit: Move L283-287 before this if to avoid some code dup.

Reply:
Fixed.